plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go

```diff
@@ -117,18 +117,35 @@ func (a *YandexCloudMonitoring) Close() error {
 	return nil
 }
 
+func setValue(value any) (float64, error) {
+	switch v := value.(type) {
+	case float64:
+		return v, nil
+	case int64:
+		return float64(v), nil
+	default:
+		return float64(-1), fmt.Errorf("expected float64, received %T: '%s'", value, value)
+	}
+}
+
 // Write writes metrics to the remote endpoint
 func (a *YandexCloudMonitoring) Write(metrics []telegraf.Metric) error {
 	var yandexCloudMonitoringMetrics []yandexCloudMonitoringMetric
 	for _, m := range metrics {
 		for _, field := range m.FieldList() {
+			value, err := setValue(field.Value)
```

Comment:
You might want to use `ToFloat64()` of the [internal package](https://github.com/influxdata/telegraf/blob/497297a92aa927ea4d0fffc2e88d10f7e6d87027/internal/type_conversions.go#L48), but not a must.

Reply:
oh hey look at that! I'll use that